kernels/volk/volk_32fc_x2_s32fc_multiply_conjugate_32fc.h

```diff
@@ -0,0 +1,247 @@
+ *
+ * \b Inputs
+ * \li aVector: The input vector to be added.
+ * \li bVector: The input vector to be conjugate and multiplied.
+ * \li scalar The complex scalar to multiply against conjugated bVector.
+ * \li num_points: The number of complex values in aVector.
+ *
+ * \b Outputs
+ * \li cVector: The vector where the results will be updated.
+ *
+ * \b Example
+ *
+ * \code
+ *
+ * \endcode
+ */
+
+#ifndef INCLUDED_volk_32fc_x2_s32fc_multiply_conjugate_32fc_a_H
+#define INCLUDED_volk_32fc_x2_s32fc_multiply_conjugate_32fc_a_H
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <volk/volk_complex.h>
+#include <float.h>
+
+
+#ifdef LV_HAVE_AVX
+#include <immintrin.h>
+
+static inline void volk_32fc_x2_s32fc_multiply_conjugate_32fc_a_avx(lv_32fc_t* cVector, const lv_32fc_t* aVector, const lv_32fc_t* bVector, const lv_32fc_t scalar, unsigned int num_points) {
+    unsigned int number = 0;
+    unsigned int i = 0;
+    const unsigned int quarterPoints = num_points / 4;
+    unsigned int isodd = num_points & 3;
+
+    __m256 x, sl, sh, y, z, tmp1, tmp2;
+
+    const lv_32fc_t* a = aVector;
+    const lv_32fc_t* b = bVector;
+    lv_32fc_t* c = cVector;
+
+    __m256 conjugator = _mm256_setr_ps(0, -0.f, 0, -0.f, 0, -0.f, 0, -0.f);
+
+    // Set up constant scalar vector
+    sl = _mm256_set1_ps(lv_creal(scalar));
+    sh = _mm256_set1_ps(lv_cimag(scalar));
+
+    for(;number < quarterPoints; number++) {
+        x = _mm256_load_ps((float*)b);
+        y = _mm256_load_ps((float*)a);
+
+        x = _mm256_xor_ps(x, conjugator);
+
+        tmp1 = _mm256_mul_ps(x,sl);
+
+        x = _mm256_shuffle_ps(x,x,0xB1);
+
+        tmp2 = _mm256_mul_ps(x,sh);
+
+        z = _mm256_addsub_ps(tmp1,tmp2);
+
+        z = _mm256_add_ps(y, z);
+
+        _mm256_store_ps((float*)c,z);
+
+        a += 4;
+        b += 4;
+        c += 4;
+    }
+
+    for(i = num_points-isodd; i < num_points; i++) {
+        *c++ = (*a++) + lv_conj(*b++) * scalar;
+    }
+
+}
+#endif /* LV_HAVE_AVX */
+
+
+#ifdef LV_HAVE_SSE3
+#include <pmmintrin.h>
+
+static inline void volk_32fc_x2_s32fc_multiply_conjugate_32fc_a_sse3(lv_32fc_t* cVector, const lv_32fc_t* aVector, const lv_32fc_t* bVector, const lv_32fc_t scalar, unsigned int num_points) {
+    unsigned int number = 0;
+    const unsigned int halfPoints = num_points / 2;
+
+    __m128 x, sl, sh, y, z, tmp1, tmp2;
+
+    const lv_32fc_t* a = aVector;
+    const lv_32fc_t* b = bVector;
+    lv_32fc_t* c = cVector;
+
+    __m128 conjugator = _mm_setr_ps(0, -0.f, 0, -0.f);
+
+    // Set up constant scalar vector
+    sl = _mm_set_ps1(lv_creal(scalar));
+    sh = _mm_set_ps1(lv_cimag(scalar));
+
+    for(;number < halfPoints; number++){
+        x = _mm_load_ps((float*)b);
+        y = _mm_load_ps((float*)a);
+
+        x = _mm_xor_ps(x, conjugator);
+
+        tmp1 = _mm_mul_ps(x,sl);
+
+        x = _mm_shuffle_ps(x,x,0xB1);
+
+        tmp2 = _mm_mul_ps(x,sh);
+
+        z = _mm_addsub_ps(tmp1,tmp2);
+
+        z = _mm_add_ps(y, z);
```

Comment:
Could you move these lines to `include/volk_sse3_intrinsics.h`? Also, that would make it easier to add an unaligned version too. If you could add an unaligned version, that would be great.

Reply:
There is a same question as avx's comment.